### PR TITLE
Removes duplicated Retangle, Oval, Diamonds Fixes #4791

### LINKED ISSUE #4791: Request in QGIS ([processing][needs-docs] port Rectangle,ovals,diamonds algorithm to C++)
### Request for documentation
From pull request QGIS/qgis#33794
Author: @alexbruy
QGIS version: 3.12

**[processing][needs-docs] port Rectangle,ovals,diamonds algorithm to C++**

## Description
Port "Rectangle, ovals, diamonds" algorithm to C++. Also make width, height and rotation parameters dynamic. 

Old Python implementations (using fixed and variable distance) marked as deprecated to maintain API compatibility.

## Checklist
- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `Fixes #11111` at the bottom of the commit message
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit
- [ ] I have evaluated whether it is appropriate for this PR to be backported, backport requests are left as label or comment

### Commits tagged with [need-docs] or [FEATURE]
"[processing]\n\n\n\n port Rectangle,ovals,diamonds algorithm to C++\nand make width, height and rotation parameters dynamic.\n\nOld Python implementations marked as deprecated to maintain API\ncompatibility."

### PR DESCRIPTION
**Pull request by SrNetoChan**
Removes duplication of the Rectangle , Oval, diamonds, which now as dynamic parameters.

Fixes #4791 

Goal:

Ticket(s): #

- [ ] Backport to LTR documentation is required

### CODE REVIEW

**Comment by SrNetoChan** [comment]
Marked as a draft to postpone the merge until we branch 3.10 Docs. Ready for reviews.

**Comment by havatv** [comment]
Covered by #4789.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 1]
```diff
@@ -4642,9 +4641,9 @@ Outputs
      - The output multipart vector layer
 
 
-.. _qgisrectanglesovalsdiamondsfixed:
+.. _qgisrectanglesovalsdiamonds:
 
-Rectangles, ovals, diamonds (fixed)
+Rectangles, ovals, diamonds
 -----------------------------------
 Creates a buffer area for all the features in an input layer with
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
Creates a buffer area with a rectangle, oval or diamond shape
for all the features of an input point layer.
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -4642,9 +4641,9 @@ Outputs
...
 different shape choice.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 1 above)
Comment (SrNetoChan) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: I like the suggestion, but since we are at it, how about "for each feature of"?

Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Updated.

**Comment by DelazJ** [comment]
Is that not the other (variable) screenshot that should be kept?

**Comment by SrNetoChan** [comment]
> Is that not the other (variable) screenshot that should be kept?

Uhmm... We could even leave both images. What do you think?

**Comment by havatv** [comment]
> Uhmm... We could even leave both images. What do you think?

Yes, both are useful.

**Comment by DelazJ** [comment]
Wouldn't that be too much for a not so central/important/complex (imho) algorithm? Maybe a mix of the two could look better than showing the two but, to me the variable one already contains implicitly the fixed one and I'm not sure it's worth the effort.

**Comment by SrNetoChan** [comment]
I have kept the variable shapes image only. I also changed the description a bit to make the dynamic nature more obvious. Please review. Thanks.

**Comment by havatv** [comment]
Thanks, @SrNetoChan!

**Pull request merged**